src/main/java/org/myrobotlab/framework/Service.java

```diff
@@ -176,6 +178,19 @@ public abstract class Service<T extends ServiceConfig> implements Runnable, Seri
    * for promoting portability and good pathing
    */
   final transient protected String ps = File.pathSeparator;
+  
+  /**
+   * queue of recent broadcasted status - usually info, warn and error.
+   * the queue's max size is currently hardcoded to 10.  I waffled a bit on
+   * it being static or not. ie Should it be a global one managed by runtime,
+   * or individual ones for each service.  Since each status contains info
+   * from which service it came from, having them non-static seems redundant.
+   * From the UI perspective its more simple to manage a single one. So,
+   * for now we'll make it static. I chose the non thread safe LinkedList for
+   * performance, and currently there is no iteration over the queue, so 
+   * concurrent modification exceptions should not occur.
+   */
+  static final protected Queue<Status> statusList = new LinkedList<>();
```

Comment:
ArrayBlockingQueue might have better performance due to cache coherency, linked lists require an additional layer of indirection and the array queue is optimized such that no moving of elements is required for removing the head. But, it's probably a micro optimization in the grand scheme of things so really doesn't matter that much